2.0.0

tristen opened the pull request:
- Pass `mode` & `colorAttribute` as optional properties to the component.
- Add `mode` & `colorAttribute` values to `onChange`.
- Remove localstorage management of color mode and attributes.
- Prefix general classnames with `cp` 
- Simplify the demo page by pointing to docs in the reame
- Tests. Close #3

tristen committed:
Ditch localStorage + pass tab and color from props

- Closes #35
- Ref #41

tristen committed:
Its an example page so log the result of `onChange` in the console

tristen committed:
Handle display color

- Also, Use `babel-plugin-object-assign` instead. Ref #15

tristen committed:
Pass `mode` & `colorAttribute` values to onChange

- Ref #41

tristen committed:
Add additions to changelog

tristen committed:
Develop via webpack

tmcw commented on index.js:
```diff
@@ -26,57 +26,63 @@ module.exports = React.createClass({
 
+  emitOnChange: function(change) {
+    var { color, mode, colorAttribute } = this.state;
+    this.props.onChange(Object.assign(
```
this library already includes xtend. why this too?

tmcw commented on .babelrc:
```diff
@@ -0,0 +1,5 @@
+{
+  "stage": 0,
+  "loose": "all",
+  "plugins": ["object-assign"]
```
Do not polyfill globals in libraries.

tmcw commented on index.js:
```diff
@@ -335,33 +341,33 @@ module.exports = React.createClass({
                     step={1} />
                 </fieldset>
-                <fieldset className={(colorMode === 'v') ? 'active' : ''}>
+                <fieldset className={(colorAttribute === 'v') ? 'active' : ''}>
```
unnecessary parens

tmcw commented on example/index.js:
```diff
@@ -28,6 +30,8 @@ var App = React.createClass({
   },
 
   onChange: function(color) {
+    console.log('onChange: ', color);
```
remove this

tristen replied:
It's on the example page and helpful to see what the onChange output looks like when evaluating this component no?

tmcw commented on index.js:
```diff
@@ -162,32 +162,34 @@ module.exports = React.createClass({
-    this.setState({mode: e.target.value});
+  setColorAttribute: function(attribute) {
+    var obj = { colorAttribute: attribute };
+    this.setState(obj);
```
setState is async and is treated here and in many other places as sync.

tmcw commented:
I don't understand what's the point of switching from xtend to Object.assign and from browserify to webpack. The problem that we have with react-colorpickr is, and has always been, robustness, and this PR does not contribute to that goal at all: this is still an untested component.

tristen committed:
Remove unnecessary parens

tristen committed:
Revert on Object.assign and use xtend

tristen committed:
Remove console.log from demo to demonstrate onChange output

tristen committed:
Merge branch 'mb-pages' into 2.0.0

tristen committed:
Prefix general class names to ensure uniquness

tristen committed:
Divide that value

tristen committed:
Merge branch 'mb-pages' into 2.0.0

tristen committed:
Add additional tests. Stub Simulate.click on tab'

tristen committed:
Remove _onClick handler that prevents bubbling from the parent. If this is even needed it should come from the implementor anyways

tristen committed:
Test HSV controls are present

tristen committed:
:page_with_curl:

tristen committed:
Focus input tests + prevent empty value from being recorded when entering an atttribute field

tristen committed:
Tests for options passed and deepEqual what onChange returns

tristen committed:
Simplify the example more.

Noting from the previous commit.

- Remove state handling and passing and explicit value
- Point to docs on GitHub. Dont repeat it here
- Simplier usage
- Pass the only required prop: onChange
